File: c/space-age/space_age.c
#include "space_age.h"

float age(planet_t planet, int64_t seconds)
{
    float earth_years = (float)seconds / 31557600.0F;

    switch (planet)
    {
    case MERCURY:
        return earth_years / 0.2408467F;
    case VENUS:
        return earth_years / 0.61519726F;
    case EARTH:
        return earth_years;
    case MARS:
        return earth_years / 1.8808158F;
    case JUPITER:
        return earth_years / 11.862615F;
    case SATURN:
        return earth_years / 29.447498F;
    case URANUS:
        return earth_years / 84.016846F;
    case NEPTUNE:
        return earth_years / 164.79132F;
    default:
        return -1;
    }
}

File: zig/space-age/space_age.zig
pub const Planet = enum {
    mercury,
    venus,
    earth,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,
    

    pub fn age(self: Planet, seconds: usize) f64 {
        const earth_year_seconds = 31557600.0;
        const float_seconds: f64 = @floatFromInt(seconds);
        const earth_years = float_seconds / earth_year_seconds;
        return switch (self) {
            .mercury => earth_years / 0.2408467,
            .venus => earth_years / 0.61519726,
            .earth => earth_years,
            .mars => earth_years / 1.8808158,
            .jupiter => earth_years / 11.862615,
            .saturn => earth_years / 29.447498,
            .uranus => earth_years / 84.016846,
            .neptune => earth_years / 164.79132,
        };
    }
};
